server/protocols/webadmin/webadmin-mail-over-web/src/test/resources/json/mail.json

```diff
@@ -0,0 +1,13 @@
+{
```

Comment:
Email is a (really) complex thing, and defining a standard JSON document  to represent an email is harder than when you propose here. To give you what it could look like you can have a look at the JMAP specification Email object (https://jmap.io/spec-mail.html#emails) and emailSubmission object (https://jmap.io/spec-mail.html#email-submission), as it serve a similar purpose.

Note that this simplistic format might work for you but might be too restrictive for other users.

Hence I am reluctant to add yet another non standard JSON representation of an email.

Also, emails size can be too big to be represented by a JSON object.

Finally you do not distinguish the mail enveloppe from the mail headers in your format.

Reply:
Thanks for the pointers. I'll start with fixing the coding style issues right away, and moving the code to the mailqueue module. After that i will try to implement the request payload as the emailSubmission object.

Reply:
> After that i will try to implement the request payload as the emailSubmission object.

Don't. I points it to show you this is a complex topic.

I would prefer passing the mail envelope via URL query parameters, and having the request payload being the raw MIME message. (I think having the raw MIME EML submission format is really easy to use). The client system could be as simple as a curl command (good inter-operability, but client system have to generate mime messages itself).

If we want to support another format, JSON based, we could use the Accept header to specify the Mail to be sent, and could be friendlier to modern application (but we won't have a one format fit all use cases without some over-engineering - we should rather start with something simple). The client system would need to be adapted to this format (limited inter-operability).

I propose you write  an API proposition, with examples, in the JIRA  ticket before implementing more things.

(You can of course take care of stylish and location issues).

Reply:
I agree that defining new formats is almost always a bad idea: it reduces the value of the feature by preventing interoperability with others tools, it will probably have new design flaws (existing formats have know design flow and people are able to deal with them).
So I would rather send the mail as a `message/rfc822` payload.